Demo/Demo/SwiftUIComponents/CheckBoxView.swift

```diff
@@ -0,0 +1,29 @@
+import SwiftUI
+
+struct CheckBoxView: View {
```

Comment:
Are you able to add a screen shot of what this looks like in the PR description?

Reply:
Sure. This is a draft so I wanted to check with Steven to see if this is what we want.
My textfield for vault customer id only appears if vault option is checked.
I put this here to make it easier for me to test vaulting result but I will put it up.
It'd be helpful to have feedback with WIP

Reply:
Looking through Apple's [human interface guidelines](https://developer.apple.com/design/human-interface-guidelines/toggles#Best-practices) it seems to indicate that unless used elsewhere in our design the preference is to use toggles when choosing 2 opposing values. It seems like this would be an appropriate use case for a toggle and would follow their guidelines for this type of UI. I think like other platform specific things we should lean into the design guidelines for each platform where possible.

Reply:
![Simulator Screen Shot - iPhone 14 Pro - 2023-06-14 at 06 13 32](https://github.com/paypal/iOS-SDK/assets/9273272/ab72bf19-319c-481d-9a88-39a8b8ede26d)

Reply:
I saw a checkbox on the Android app, so I was trying to replicate with toggle option on the checkbox.

Reply:
Addressed in commit a1d9a4758c29e5f82cd6fb3234b7332427433c4f